packages/compass-import-export/src/modules/export.js

```diff
@@ -426,70 +439,73 @@ export const startExport = () => {
 
     let exportSucceded = true;
 
-    const spec = exportData.isFullCollection
-      ? { filter: {} }
-      : exportData.query;
-    const numDocsToExport = exportData.isFullCollection
-      ? await fetchDocumentCount(dataService, ns, spec)
-      : exportData.count;
-    // filter out only the fields we want to include in our export data
-    const projection = Object.fromEntries(
-      Object.entries(exportData.fields).filter(
-        (keyAndValue) => keyAndValue[1] === 1
-      )
-    );
-    if (
-      Object.keys(projection).length > 0 &&
-      (undefined === exportData.fields._id || exportData.fields._id === 0)
-    ) {
-      projection._id = 0;
-    }
-    log.info(
-      mongoLogId(1001000083),
-      'Export',
-      'Start reading from collection',
-      {
-        ns,
-        numDocsToExport,
-        spec,
-        projection,
-      }
-    );
-    const progress = createProgressStream({
-      objectMode: true,
-      length: numDocsToExport,
-      time: 250 /* ms */,
-    });
-
-    progress.on('progress', function (info) {
-      dispatch(onProgress(info.percentage, info.transferred));
-    });
-
-    log.info(mongoLogId(1001000084), 'Export', 'Start writing to file', {
-      ns,
-      fileType: exportData.fileType,
-      fileName: exportData.fileName,
-      fields: Object.keys(exportData.allFields),
-    });
-    // Pick the columns that are going to be matched by the projection,
-    // where some prefix the field (e.g. ['a', 'a.b', 'a.b.c'] for 'a.b.c')
-    // has an entry in the projection object.
-    const columns = Object.keys(exportData.fields).filter((field) =>
-      field
-        .split('.')
-        .some(
-          (_part, index, parts) =>
-            projection[parts.slice(0, index + 1).join('.')]
+    const dest = fs.createWriteStream(exportData.fileName);
+    let columns, source, numDocsToExport;
+    let numDocsActuallyExported = 0;
+    if (!isEmpty(exportData.aggregation)) {
```

Comment:
How do you feel about breaking the code in these condition blocks into functions? I'm thinking it might improve readability.

Maybe something like 
```ts
const {
  columns,
  source,
  numDocsToExport
} = getAggregationExportSource(dataService, ns, exportData);
```